src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -81,6 +81,11 @@ public function onKernelRequest(GetResponseEvent $event)
         if (!$event->isMasterRequest() || '/' !== $request->getPathInfo()) {
             return;
         }
+        // Ignore requests from referrers with the same HTTP host in order to prevent
+        // changing language for users who possibly already selected it for this application.
+        if (0 === stripos($request->server->get('HTTP_REFERER'), $request->getSchemeAndHttpHost())) {
```

Comment:
:+1: `$request->headers->get('referer')` for three reasons:
* `$request->headers` taken from the `$_SERVER`
* `HTTP_REFERER` is a request-header field: http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
* Preferred by Symfony core: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/EventListener/RouterListener.php#L119, https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationSuccessHandler.php#L122